src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -110,7 +110,7 @@ public HashSet(IEnumerable<T> collection, IEqualityComparer<T>? comparer) : this
 
                 UnionWith(collection);
 
-                if (_count > 0 && _entries!.Length / _count > ShrinkThreshold)
+                if (IsTrimNeeded())
```

Comment:
This looks like an unrelated change. Also, why does this need to be extracted to a helper method?

Reply:
This check used in Initialization (when hashset created with another collection) and in ConstructFrom. I moved this to method to unify this check